src/Modules/UngradedModules.jl

```diff
@@ -3866,15 +3914,18 @@ end
 Compute the `i`-th homology of `C`.
 """
 function homology(C::Hecke.ChainComplex{ModuleFP}, i::Int)
-  @assert length(C) > 0 #TODO we need actually only the base ring
-  if i == 0
-    return kernel(map(C,1))[1]
-  elseif i == length(C)
-    return quo(obj(C,i),image(map(C,i))[1])
-  elseif i < 0 || i > length(C)
-    return FreeMod(base_ring(obj(C,1)),0)
+  chain_range = range(C)
+  @assert length(chain_range) > 0 #TODO we need actually only the base ring
+  if i == first(chain_range)
+    return kernel(map(C,i))[1]
+  elseif i == last(chain_range)
+    f = map(C,i)
+    return quo(codomain(f),image(f)[1])
+  elseif i in chain_range
+    next_index = C.direction == :left ? i-1 : i+1
```

Comment:
I would suggest not to access members of `C` directly if possible, this mixes details of the implementation with the interface. The proper way to do this is
```suggestion
    next_index = is_chain_complex(C) ? i-1 : i+1
```

That said... it seems awkward that one has to do such a thing. Perhaps we should have an API for the chain complexes to ask for "the next map after this one" which encapsulates this kind of stuff? Perhaps one already exists, but I didn't find it. Perhaps @thofma and @fieker know or have thoughts